bin/network-monitor/src/remote_prover.rs

```diff
@@ -0,0 +1,318 @@
+
+/// Remote prover types
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum ProofType {
+    Transaction,
+    Block,
+    Batch,
+}
+
+impl From<ProofType> for proto::remote_prover::ProofType {
+    fn from(value: ProofType) -> Self {
+        match value {
+            ProofType::Transaction => proto::remote_prover::ProofType::Transaction,
+            ProofType::Block => proto::remote_prover::ProofType::Block,
+            ProofType::Batch => proto::remote_prover::ProofType::Batch,
+        }
+    }
+}
+
+impl From<proto::remote_prover::ProofType> for ProofType {
+    fn from(value: proto::remote_prover::ProofType) -> Self {
+        match value {
+            proto::remote_prover::ProofType::Transaction => ProofType::Transaction,
+            proto::remote_prover::ProofType::Batch => ProofType::Batch,
+            proto::remote_prover::ProofType::Block => ProofType::Block,
+        }
+    }
+}
+
+// REMOTE PROVER TEST TYPES
+// ================================================================================================
+
+/// Details of a remote transaction prover test.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ProverTestDetails {
+    pub test_duration_ms: u64,
+    pub proof_size_bytes: usize,
+    pub success_count: u64,
+    pub failure_count: u64,
+    pub proof_type: ProofType,
+}
+
+// REMOTE TRANSACTION PROVER TEST TASK
+// ================================================================================================
+
+/// Runs a task that continuously tests remote prover functionality and updates a watch channel.
+///
+/// This function spawns a task that periodically sends mock request payloads to a remote prover
+/// and measures the success/failure rate and performance metrics for proof generation.
+///
+/// # Arguments
+///
+/// * `prover_url` - The URL of the remote prover service to test.
+/// * `name` - The name of the remote prover.
+/// * `proof_type` - The type of proof to test.
+/// * `serialized_request_payload` - The serialized request payload to send to the remote prover.
+/// * `status_sender` - The sender for the watch channel.
+///
+/// # Returns
+///
+/// `Ok(())` if the task completes successfully, or an error if the task fails.
+#[instrument(target = COMPONENT, name = "remote-prover-test-task", skip_all)]
+pub async fn run_remote_prover_test_task(
+    prover_url: Url,
+    name: &str,
+    proof_type: ProofType,
+    serialized_request_payload: proto::remote_prover::ProofRequest,
+    status_sender: watch::Sender<ServiceStatus>,
+) -> anyhow::Result<()> {
+    let mut client = ClientBuilder::new(prover_url)
+        .with_tls()
+        .expect("TLS is enabled")
+        .with_timeout(Duration::from_secs(10))
+        .without_metadata_version()
+        .without_metadata_genesis()
+        .connect_lazy::<RemoteProverClient>();
+
+    let mut interval = tokio::time::interval(Duration::from_secs(30)); // Test every 30 seconds
```

Comment:
Just wondering if we should make this relatively large - say 1 or 2 minutes? Particularly if we expect testnet/mainnet provers to be under significant load (I'm not sure how much load this request itself involves).

And it would be good to make this configurable but not required for this PR.

Reply:
Yes, agree that the best is making this configurable. The configuration options is growing and I'm thinking that switching to `clap` might be a good idea, wdyt?

Reply:
In the meantime, I changed it to 2 mins.